readthedocs/projects/forms.py

```diff
@@ -211,6 +212,59 @@ class Meta(object):
             # 'num_major', 'num_minor', 'num_point',
         )
 
+    def __init__(self, *args, **kwargs):
+        super(ProjectAdvancedForm, self).__init__(*args, **kwargs)
+
+        # default_version ChoiceField
+        self.versions_qs = self.instance.versions.all()
+        self.active = self.versions_qs.filter(active=True)
+        if self.active.exists():
+            self.version_choices = [(version.slug, version.verbose_name) for version in self.active]
+            self.fields['default_version'] = forms.ChoiceField(
```

Comment:
I think we should use the `widgets` attribute instead of re-define a field: https://docs.djangoproject.com/en/2.0/topics/forms/modelforms/#overriding-the-default-fields

Reply:
In fact, the same should have used for `python_interpreter` from this form since it just changes the widget and adds an initial value.

Reply:
Do we need initial? If we have some default in the model, we don't need to set initial in form.